contracts/dao/Dao.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.8.4;
+
+import "../interfaces/dao/IDao.sol";
+
+// Provides functions to be used internally to track proposals and votes
+abstract contract DAO is IDAO {
+  mapping(uint256 => ProposalMetaData) private _proposals;
+  uint256 private _nextProposalId;
+
+  function getProposalCreator(uint256 id) external view override returns (address) {
+    return _proposals[id].creator;
+  }
+  function getProposalInfo(uint256 id) external view override returns (string memory) {
+    return _proposals[id].info;
+  }
+  function getVoteStatus(uint256 id, address voter) external view override returns (bool) {
+    return _proposals[id].voters[voter];
+  }
+  function getTimeSubmitted(uint256 id) external view override returns (uint256) {
+    return _proposals[id].submitted;
+  }
+  function getTimeExpires(uint256 id) external view override returns (uint256) {
+    return _proposals[id].expires;
+  }
+  function getCompleted(uint256 id) external view override returns (bool) {
+    return _proposals[id].completed;
+  }
+
+  modifier activeProposal(uint256 id) {
+    // Proposal must actually exist
+    require(_proposals[id].submitted != 0);
+    // Has yet to expire
+    require(block.timestamp < _proposals[id].expires);
+    // Wasn't completed
+    require(!_proposals[id].completed);
+    _;
+  }
+
+  // Should only be called by a function which attaches coded meaning to this metadata
+  function _createProposal(string calldata info, uint256 expires) internal returns (uint256 id) {
```

Comment:
Currently info is a blob; do we plan to break that out in this release?

Reply:
How so? This is solely the metadata, so the info string would be "Hire X to paint the house red.". It's the actual Proposal which would specify payment and handle those variables.